includes/class-edd-db-customers.php

```diff
@@ -257,6 +259,50 @@ public function decrement_stats( $customer_id = 0, $amount = 0.00 ) {
 	}
 
 	/**
+	 * Updates the email address of a customer record when the email on a user is updated
+	 *
+	 * @access  public
+	 * @since   2.4
+	*/
+	public function update_customer_email_on_user_update( $user_id = 0, $old_user_data ) {
+
+		$customer = new EDD_Customer( $user_id, true );
+
+		if( ! $customer ) {
+			return false;
+		}
+
+		$user = get_userdata( $user_id );
+
+		if( $user->user_email !== $customer->email ) {
```

Comment:
$user should not be assumed to be a user object here. We should ensure it is not empty, that it is an object and that it is a user object before assuming it is a user object (it could be int 0, int 0, or a WP error object respectively)